woocommerce/woocommerce-gateway-paypal-express-checkout
# Add BLIK and MercadoPago as funding methods

**Issue**: #796

---

### Description

From the current list of [alternative payment methods](https://developer.paypal.com/docs/checkout/integration-features/alternative-payment-methods/#availability), we were missing both "BLIK" and "MercadoPago".
This PR adds both options as funding methods and also sorts the dropdown options alphabetically, leaving only PayPal Credit and Debit/Credit Card at the top, as those are the ones more commonly used.

### Steps to test:


1. Check out `trunk`.
1. Add an item to the cart and go to the checkout page.
1. Use the following snippet to temporarily change your site to PLN and your buyer country to Poland:
   ```php
   add_filter( 'woocommerce_paypal_express_checkout_sdk_script_args', function( $args ) {
	   $args['buyer-country'] = 'PL';
	   return $args;
   } );
   
   add_filter( 'woocommerce_currency', function( $currency ) {
	   return 'PLN';
   } );
   ```
1. Confirm that BLIK and P24 are available as payment options:
   <img width="374" alt="Screen Shot 2021-05-27 at 10 27 23" src="https://user-images.githubusercontent.com/184724/119836060-06fdd580-bec7-11eb-98d4-87d3a543603a.png">
1. Go to WC > Settings > Payments > PayPal Checkout > Hide Funding Method(s) and confirm that BLIK is not an option.
1. Check out this branch (`issue/796`) and see an option for "BLIK" above. Choose to hide the funding method.
1. Refresh the checkout page and confirm that BLIK is no longer an option:
   <img width="348" alt="Screen Shot 2021-05-27 at 10 25 14" src="https://user-images.githubusercontent.com/184724/119836315-42000900-bec7-11eb-8379-4a7dd202a5cb.png">


### Changelog entry
- Add BLIK and MercadoPago as funding methods

Closes #796.
